vncviewer/GestureHandler.h

```diff
@@ -0,0 +1,97 @@
+/* Copyright 2019 Aaron Sowry for Cendio AB
+ * 
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * 
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#ifndef __GESTUREHANDLER_H__
+#define __GESTUREHANDLER_H__
+
+#include <map>
+
+#include <X11/Xlib.h>
+#include <X11/extensions/XInput2.h>
+
+#include <rfb/Timer.h>
+
+// Internal state bitmasks
+#define GH_NOGESTURE   0
+#define GH_LEFTBTN     1
+#define GH_MIDDLEBTN   2
+#define GH_RIGHTBTN    4
+#define GH_VSCROLL     8
+#define GH_HSCROLL     16
```

Comment:
Any reason why we are separating these? It means users will have to keep switching back and forth if they need to pan in both directions.

Reply:
I guess it's a limitation of the messaging in `GestureHandler`. A gesture update specifies a magnitude which only works for a single dimension, not two.

Also I think combining them would be confusing in practice unless we added the ability to "scroll" diagonally too, which would be hell to implement. "Free" panning like that is really catered for by a click-and-drag if the application supports it.

Reply:
It's new code, so it's a self imposed limitation. :)

Wouldn't diagonally "just work" since we would send one horizontal and one vertical scroll for such a gesture?

Reply:
If we combined them, we'd only have one gesture type for that action. So we would need to send a `GH_GestureUpdate` of type `GH_PAN` with both `x` and `y` magnitudes, which would mean doing this for all gestures, even ones where it doesn't make much sense.

It's quite a lot of work for no extra functionality since free-panning is already catered for in most applications by click-and-drag (either left or middle button, I've found). Perhaps we can add this as an enhancement later?

Reply:
This is resolved - diagonal panning works.